streamy.py

```diff
@@ -75,3 +56,60 @@ def stream(fp, json_lines=False, **kwds):
             obj = json.loads(json_str, **kwds)
             start_pos += end_pos
             yield obj
+
+
+def _stream_without_seek(fp, kwds):
+    buf = ''
+    # XXX: alternatively could read a limited size buffer
```

Comment:
Can you explain this comment a little more (in this review)?  Then we can decide either to do that, or to do what's in the review, and dump the comment so later readers don't have to see it.  :-)

Reply:
If a single line contains many JSON, this may be consuming more memory than appropriate. Instead could use a `readline(BUF_SIZE)` loop so that no memory is wasted, but parse attempts are wasted on line JSON blobs on a single line. So it's a tradeoff. Hence the on my initial implementation rather than prematurely optimising.